src/Stratis.Bitcoin/Base/TimeSyncBehavior.cs

```diff
@@ -296,6 +291,29 @@ private void RecalculateTimeOffsetLocked()
 
             this.logger.LogTrace("(-):{0}={1}", nameof(this.timeOffset), this.timeOffset.TotalSeconds);
         }
+         
+        /// <summary>
+        /// Gets the median offset of all the inbound samples 
+        /// limited to 1 less than the <see cref="OutboundToInboundWeightRatio"/> inbound per outbound
+        /// and the outbound samples with a weighting of the OutboundToInboundWeightRatio.
```

Comment:
I think this comment is a bit misleading. After reading it I'd expect (assuming I have 1 outbound and 40 inbound connections) that the algho will select 1 outbound and 9 inbound ones and then calculate the median. 

So I suggest using something like: 

Calculates median offset using all inbound samples and OutboundToInboundWeightRatio copies of the outbound samples (which is pretty much the same as the comment that was removed for some reason).

Reply:
>After reading it I'd expect (assuming I have 1 outbound and 40 inbound connections) that the algho will select 1 outbound and 9 inbound ones and then calculate the median.

Yes that is correct and intended.  

Discussion continued on the thread below with your alternative implementation suggestion